azure TestGetInstanceIDByNodeName data race

/kind bug

**What happened**:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/58320/pull-kubernetes-bazel-test/25992/
```
bazel test //pkg/cloudprovider/providers/azure:go_default_test
E0116 08:02:57.246748      14 azure_loadbalancer.go:288] selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(true) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])
E0116 08:02:57.345730      14 azure_loadbalancer.go:288] selectLoadBalancer: cluster(testCluster) service(default/service-2) isInternal(false) -  all available load balancers have exceeded maximum rule limit 1, vmSetNames ([as])
E0116 08:02:58.790605      14 azure_blobDiskController.go:76] azureDisk - getAllStorageAccounts error: azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/--subscription-id--/resourceGroups/--resource-group--/providers/Microsoft.Storage/storageAccounts?api-version=2017-06-01: StatusCode=0 -- Original Error: adal: Refresh request failed. Status Code = '400'. Response body: {"error":"invalid_request","error_description":"AADSTS90002: Requested tenant identifier '--tenant-id--' is not valid.\r\nTrace ID: a6d564a2-94fc-432c-a8a0-23f026682500\r\nCorrelation ID: 925ff6f8-26c0-479c-bab9-2eff776147e4\r\nTimestamp: 2018-01-16 08:02:59Z","error_codes":[90002],"timestamp":"2018-01-16 08:02:59Z","trace_id":"a6d564a2-94fc-432c-a8a0-23f026682500","correlation_id":"925ff6f8-26c0-479c-bab9-2eff776147e4"}
E0116 08:02:58.793439      14 azure_blobDiskController.go:76] azureDisk - getAllStorageAccounts error: storage.AccountsClient#ListByResourceGroup: Invalid input: autorest/validation: validation failed: parameter=resourceGroupName constraint=MinLength value="" details: value length must be greater than or equal to 1
E0116 08:02:58.795075      14 azure_blobDiskController.go:76] azureDisk - getAllStorageAccounts error: storage.AccountsClient#ListByResourceGroup: Invalid input: autorest/validation: validation failed: parameter=resourceGroupName constraint=MinLength value="" details: value length must be greater than or equal to 1
E0116 08:02:58.866717      14 azure_blobDiskController.go:76] azureDisk - getAllStorageAccounts error: azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/--subscription-id--/resourceGroups/--resource-group--/providers/Microsoft.Storage/storageAccounts?api-version=2017-06-01: StatusCode=0 -- Original Error: adal: Refresh request failed. Status Code = '400'. Response body: {"error":"invalid_request","error_description":"AADSTS90002: Requested tenant identifier '--tenant-id--' is not valid.\r\nTrace ID: b601ebed-6bfd-496c-a9da-ad8286413200\r\nCorrelation ID: 508095ad-5b26-4faf-8a0f-fef89cd8b0f0\r\nTimestamp: 2018-01-16 08:02:59Z","error_codes":[90002],"timestamp":"2018-01-16 08:02:59Z","trace_id":"b601ebed-6bfd-496c-a9da-ad8286413200","correlation_id":"508095ad-5b26-4faf-8a0f-fef89cd8b0f0"}
==================
WARNING: DATA RACE
Write at 0x00c42003b4b8 by goroutine 43:
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.setTestVirtualMachineScaleSets()
      pkg/cloudprovider/providers/azure/azure_vmss_test.go:71 +0x10d5
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.TestGetInstanceIDByNodeName()
      pkg/cloudprovider/providers/azure/azure_vmss_test.go:139 +0x366
  testing.tRunner()
      GOROOT/src/testing/testing.go:746 +0x16c

Previous read at 0x00c42003b4b8 by goroutine 40:
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.(*scaleSet).listScaleSetsWithRetry.func1()
      pkg/cloudprovider/providers/azure/azure_vmss.go:406 +0x94
  k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff()
      vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:203 +0xb0
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.(*scaleSet).listScaleSetsWithRetry()
      pkg/cloudprovider/providers/azure/azure_vmss.go:405 +0x2e9
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.(*scaleSet).updateCache()
      pkg/cloudprovider/providers/azure/azure_vmss.go:106 +0x53
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.newScaleSet.func1()
      pkg/cloudprovider/providers/azure/azure_vmss.go:96 +0x8d
  k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1()
      vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x6f
  k8s.io/apimachinery/pkg/util/wait.JitterUntil()
      vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xcd
  k8s.io/apimachinery/pkg/util/wait.Until()
      vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x5a

Goroutine 43 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      GOROOT/src/testing/testing.go:746 +0x16c
  testing.runTests()
      GOROOT/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:921 +0x206
  main.main()
      bazel-out/k8-fastbuild/bin/pkg/cloudprovider/providers/azure/go_default_test~/testmain.go:194 +0x367

Goroutine 40 (running) created at:
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.newScaleSet()
      pkg/cloudprovider/providers/azure/azure_vmss.go:92 +0x2a6
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.newTestScaleSet()
      pkg/cloudprovider/providers/azure/azure_vmss_test.go:28 +0x34
  k8s.io/kubernetes/pkg/cloudprovider/providers/azure.TestGetInstanceIDByNodeName()
      pkg/cloudprovider/providers/azure/azure_vmss_test.go:105 +0x65
  testing.tRunner()
      GOROOT/src/testing/testing.go:746 +0x16c
==================
--- FAIL: TestGetInstanceIDByNodeName (0.00s)
	testing.go:699: race detected during execution of test
FAIL

```

--- a/pkg/cloudprovider/providers/azure/azure_vmss_test.go
+++ b/pkg/cloudprovider/providers/azure/azure_vmss_test.go
@@ -24,12 +24,15 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func newTestScaleSet() *scaleSet {
-	ss := newScaleSet(getTestCloud())
+func newTestScaleSet(scaleSetName string, vmList []string) *scaleSet {
+	cloud := getTestCloud()
+	setTestVirtualMachineCloud(cloud, scaleSetName, vmList)
+	ss := newScaleSet(cloud)
+
 	return ss.(*scaleSet)
 }
 
-func setTestVirtualMachineScaleSets(ss *scaleSet, scaleSetName string, vmList []string) {
+func setTestVirtualMachineCloud(ss *Cloud, scaleSetName string, vmList []string) {
 	virtualMachineScaleSetsClient := newFakeVirtualMachineScaleSetsClient()
 	scaleSets := make(map[string]map[string]compute.VirtualMachineScaleSet)
 	scaleSets["rg"] = map[string]compute.VirtualMachineScaleSet{
@@ -63,13 +66,13 @@ func setTestVirtualMachineScaleSets(ss *scaleSet, scaleSetName string, vmList []
 			},
 			ID:         &ID,
 			InstanceID: &instanceID,
-			Location:   &ss.Cloud.Location,
+			Location:   &ss.Location,
 		}
 	}
 	virtualMachineScaleSetVMsClient.setFakeStore(ssVMs)
 
-	ss.Cloud.VirtualMachineScaleSetsClient = virtualMachineScaleSetsClient
-	ss.Cloud.VirtualMachineScaleSetVMsClient = virtualMachineScaleSetVMsClient
+	ss.VirtualMachineScaleSetsClient = virtualMachineScaleSetsClient
+	ss.VirtualMachineScaleSetVMsClient = virtualMachineScaleSetVMsClient
 }
 
 func TestGetScaleSetVMInstanceID(t *testing.T) {
@@ -102,8 +105,6 @@ func TestGetScaleSetVMInstanceID(t *testing.T) {
 }
 
 func TestGetInstanceIDByNodeName(t *testing.T) {
-	ss := newTestScaleSet()
-
 	testCases := []struct {
 		description string
 		scaleSet    string
@@ -136,7 +137,8 @@ func TestGetInstanceIDByNodeName(t *testing.T) {
 	}
 
 	for _, test := range testCases {
-		setTestVirtualMachineScaleSets(ss, test.scaleSet, test.vmList)
+		ss := newTestScaleSet(test.scaleSet, test.vmList)
+
 		real, err := ss.GetInstanceIDByNodeName(test.nodeName)
 		if test.expectError {
 			assert.Error(t, err, test.description)

